examples/utils/xfeat.py

```diff
@@ -0,0 +1,143 @@
+from typing import List
+
+import cv2
+import depthai as dai
+from visualization.visualizers import xfeat_visualizer
+
+from depthai_nodes.ml.parsers import XFeatMonoParser, XFeatStereoParser
+
+
+def xfeat_mono(nn_archive: dai.NNArchive, input_shape: List[int], fps_limit: int):
+    """Run the XFeatMonoParser on a single camera.
+
+    It lets you set the reference frame by pressing S-key.
+    """
+    previous_frame = None
+    with dai.Pipeline() as pipeline:
+        # Set up camera
+        cam = pipeline.create(dai.node.Camera).build()
+
+        # Set up the neural network
+        network = pipeline.create(dai.node.NeuralNetwork).build(
+            cam.requestOutput(
+                input_shape, type=dai.ImgFrame.Type.BGR888p, fps=fps_limit
+            ),
+            nn_archive,
+        )
+
+        # Set up parser
+        parser = XFeatMonoParser()
+        parser.setOriginalSize(input_shape)
+        parser.setInputSize(input_shape)
+        parser.setMaxKeypoints(2048)
```

Comment:
Could we, just for the example, reduce the max number of keypoints? When running the example, the visual is not very example friendly. I would rather have maybe max 25 of the most confident keypoints to showcase the model. 
![image](https://github.com/user-attachments/assets/5e7002eb-265e-447a-983d-46b4c3a83ccb)

Reply:
Yeah good point. The problem with with less keypoints is that the rectangle can not be accurately calculated. Also here I wanted to showcase that XFeat detects and matches many features compared to others (e.g. SIFT) but true visual is not great. Maybe like you said we display less keypoints in Stereo mode and leave out the rectangle caluation? But in the mono mode I would leave it as it is because when you move the camera fewer kpts will be detected anyway.

Reply:
Yes maybe leave out rectangle for stereo as its either super hard to see with all the keypoints and just showcase keypoints. For mono agree that the num keypoints gets reduceds so we can leave it as-is.